File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        N = len(nums)
        total = 0
        for j in range(k):
            total += nums[j]
        res = total / k
        for i in range(k, N):
            total += nums[i] - nums[i-k]
            res = max(res, total/k)
        return res

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.kt
class Solution {
    fun findMaxAverage(nums: IntArray, k: Int): Double {
        var total = 0.0
        for (i in 0 until k) {
            total += nums[i]
        }
        var result: Double = total / k
        for (i in k..nums.lastIndex) {
            total += nums[i] - nums[i-k]
            result = maxOf(total / k, result)
        }
        return result
    }
}